build_tools/docker/base/Dockerfile

```diff
@@ -0,0 +1,29 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM ubuntu:18.04 AS final
```

Comment:
Could you explain what you mean when you name images `AS final`? I'm not sure why sometimes the only `FROM` statement in a file gets an `AS final`, while other times it doesn't, and I can't find a reference for the convention.

Reply:
In multistage builds each FROM instruction creates a new stage. You can optionally [use `AS <name>` to name them](https://docs.docker.com/develop/develop-images/multistage-build/#name-your-build-stages). You can use this name to reference them from later stages (as I have done for the `COPY` instructions). You can also [invoke the build command with a target stage](https://docs.docker.com/develop/develop-images/multistage-build/#stop-at-a-specific-build-stage) to build up to that stage. I decided to call the last stage in each file `final` to indicate that it's the final product and so they could all be addressed as `--target final` from the build command in a uniform way.

We could leave them off, or choose a different name. Don't have strong feelings.

Reply:
That makes sense, but I don't understand why in some files like `.../cmake-python-nvidia/Dockerfile` the only from statement doesn't have an `AS final` on it, where as in `.../cmake-python-swiftshader/Dockerfile` the only from statement does:

```dockerfile
FROM gcr.io/iree-oss/cmake-python-vulkan
# vs
FROM gcr.io/iree-oss/cmake-python-vulkan AS final
```

Reply:
Because I forgot :-) Added the missing ones